akka-javasdk-tests/src/test/java/akkajavasdk/WorkflowTest.java

```diff
@@ -104,6 +104,16 @@ public void shouldTransferMoneyAndDelete() {
               assertThat(lastStep).isEqualTo("logAndStop");
             });
 
+    Awaitility.await()
+        .ignoreExceptions()
+        .atMost(5, TimeUnit.of(SECONDS))
+        .untilAsserted(
+            () -> {
+              List<String> workflowSteps =
+                  telemetryReader.getWorkflowSteps(TransferWorkflow.class, transferId);
+              assertThat(workflowSteps).containsOnly("withdraw-step", "deposit-step", "logAndStop");
```

Comment:
@pvlugter unfortunately all tracing assertions should be wrapped with Awaitility. Sometimes spans arrive later than the observable outcome, and the assertion fails.